### DISCUSSION TITLE
How to count data by groupBy query

### DISCUSSION THREAD

**Discussion by treedots-kavan** [prisma/prisma#9993] [Q&A] [answered]
Hi, I have a query that pagination groupBy query, and I want to have the total count based on groupBy query.
But I stuck with Prisma. How we can do that on Prisma?

```ts
const example = await context.prisma.example.groupBy({
        by: ['date', 'user_id'],
        orderBy: {
          date: 'desc'
        },
        take: take,
        skip: skip,
});
```

**Comment by l2D** [comment, marked as answer]
Did you try this?
```js
const example = await context.prisma.example.groupBy({
        by: ['date', 'user_id'],
        _count: {
           _all: true,
           _sum: { field_name: true }
        },
        orderBy: {
          date: 'desc'
        },
        take: take,
        skip: skip,
});
```
Output:
```js
[
  {
    date: YYYY-MM-DDTHH:mm:ss:zz,
    user_id: 1,
    _count: {
      _all: 70,
      _sum: { field_name: 100 }
    },
  },
  {
    date: YYYY-MM-DDTHH:mm:ss:zz,
    user_id: 2,
    _count: {
      _all: 50,
      _sum: { field_name: 80 }
    },
  },
]
```

**Comment by Riunshow** [comment]
I think this is the answer, but it's a bug with [https://github.com/prisma/prisma/issues/9590](https://github.com/prisma/prisma/issues/9590)
```js
  const count = await this.prismaClient.example.count({
    distinct: 'name',
  });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
